src/transformers/processing_utils.py

```diff
@@ -54,6 +68,226 @@
+        truncation (`bool`, `str` or [`~tokenization_utils_base.TruncationStrategy`], *optional*, defaults to `False`):
+            Activates and controls truncation.
+        max_length (`int`, *optional*):
+            Controls the maximum length to use by one of the truncation/padding parameters.
+        stride (`int`, *optional*, defaults to 0):
+            If set, the overflowing tokens will contain some tokens from the end of the truncated sequence.
+        is_split_into_words (`bool`, *optional*, defaults to `False`):
+            Whether or not the input is already pre-tokenized.
+        pad_to_multiple_of (`int`, *optional*):
+            If set, will pad the sequence to a multiple of the provided value.
+        return_token_type_ids (`bool`, *optional*):
+            Whether to return token type IDs.
+        return_attention_mask (`bool`, *optional*):
+            Whether to return the attention mask.
+        return_overflowing_tokens (`bool`, *optional*, defaults to `False`):
+            Whether or not to return overflowing token sequences.
+        return_special_tokens_mask (`bool`, *optional*, defaults to `False`):
+            Whether or not to return special tokens mask information.
+        return_offsets_mapping (`bool`, *optional*, defaults to `False`):
+            Whether or not to return `(char_start, char_end)` for each token.
+        return_length (`bool`, *optional*, defaults to `False`):
+            Whether or not to return the lengths of the encoded inputs.
+        verbose (`bool`, *optional*, defaults to `True`):
+            Whether or not to print more information and warnings.
+        padding_side (`str`, *optional*):
+            The side on which padding will be applied.
+    """
+
+    add_special_tokens: Optional[bool]
+    padding: Union[bool, str, PaddingStrategy]
+    truncation: Union[bool, str, TruncationStrategy]
+    max_length: Optional[int]
+    stride: Optional[int]
+    is_split_into_words: Optional[bool]
+    pad_to_multiple_of: Optional[int]
+    return_token_type_ids: Optional[bool]
+    return_attention_mask: Optional[bool]
+    return_overflowing_tokens: Optional[bool]
+    return_special_tokens_mask: Optional[bool]
+    return_offsets_mapping: Optional[bool]
+    return_length: Optional[bool]
+    verbose: Optional[bool]
+    padding_side: Optional[str]
+
+
+class ImagesKwargs(TypedDict, total=False):
+    """
+    Keyword arguments for image processing. For extended documentation, check the appropriate ImageProcessor
+    class methods and docstrings.
+
+    Attributes:
+        do_resize (`bool`, *optional*, defaults to `True`):
+            Whether to resize the image.
+        size (`Dict[str, int]`, *optional*, defaults to `{'shortest_edge': 288}`):
+            Resize the shorter side of the input to `size["shortest_edge"]`.
+        size_divisor (`int`, *optional*, defaults to 32):
+            The size by which to make sure both the height and width can be divided.
```

Comment:
Defaults for these as a processor-wide docstring aren't ideal. In particular for size which varies widely across models (if we were to have a default `{"height": 224, "width": 224}` would be better)

Do we need to have defaults in the docstring? Could we instead just list the types which should be consistent?

Reply:
Design has evolved, missed this, my bad: We don't need defaults for the docstring at all! Even, we _should_ not, because it just sets the types - TypedDicts have no purpose setting default values, hence the weird `_defaults` key from above comment.